result_test.go

```diff
@@ -140,3 +140,197 @@ func BenchmarkCreateVendorTree(b *testing.B) {
 	sm.Release()
 	os.RemoveAll(tmp) // comment this to leave temp dir behind for inspection
 }
+
+func TestStripVendor(t *testing.T) {
+	t.Run("vendor directory", stripVendorTestCase(fsTestCase{
+		before: filesystemState{
+			dirs: []fsPath{
+				fsPath{"package"},
+				fsPath{"package", "vendor"},
```

Comment:
Oh, the other reason is that it helps the before and after states look more parallel. This is the redundant version:
```go
	t.Run("nonvendor symlink", stripVendorTestCase(fsTestCase{
		before: filesystemState{
			dirs: []fsPath{
				fsPath{"package"},
				fsPath{"package", "_vendor"},
			},
			links: []fsLink{
				fsLink{
					path: fsPath{"package", "link"},
					to:   "_vendor",
				},
			},
		},
		after: filesystemState{
			dirs: []fsPath{
				fsPath{"package"},
				fsPath{"package", "_vendor"},
			},
			links: []fsLink{
				fsLink{
					path: fsPath{"package", "link"},
					to:   "_vendor",
				},
			},
		},
	}))
```

And here's the trimmed down one:
```go
	t.Run("nonvendor symlink", stripVendorTestCase(fsTestCase{
		before: filesystemState{
			dirs: []fsPath{
				fsPath{"package", "_vendor"},
			},
			links: []fsLink{
				fsLink{
					path: fsPath{"package", "link"},
					to:   "_vendor",
				},
			},
		},
		after: filesystemState{
			dirs: []fsPath{
				fsPath{"package"},
				fsPath{"package", "_vendor"},
			},
			links: []fsLink{
				fsLink{
					path: fsPath{"package", "link"},
					to:   "_vendor",
				},
			},
		},
	}))
```

I like how the after state matches the before state's directory listing, even if it's a little more verbose, so I think I'll keep these in.